### ISSUE TITLE
Request controller argument is first resolved as Doctrine entity, causing 40-50ms performance loss

### ISSUE THREAD

**Issue by Xesau** [symfony/symfony#54337]
### Symfony version(s) affected

7.0.5

### Description

For the tag `controller.argument_value_resolver`, the `argument_resolver.request` service has a priority of 50, while  `doctrine.orm.entity_value_resolver` has a priority of 110. As a result, EntityValueResolver from DoctrineBundle first tries to resolve the `Symfony\Component\HttpFoundation\Request` controller argument as an entity. This is nonsensical and will cause Doctrine to try to find an entity manager and repository for it. In my case this leads to an extra 40-50ms of Total execution time on average.
By changing the priority of `argument_resolver.request` to 150, the Request argument is resolved immediately without this detour.

### How to reproduce

Create a clean installation of `symfony/skeleton` and require `symfony/webapp-pack`. Create a controller with a `Request $request` function argument and visit it. Inspect the Performance profiler page for the request.

**Comment by MatTheCat** [comment]
You can use the `ValueResolver` attribute to specify which resolver must be called for a given parameter; see https://symfony.com/doc/current/controller/value_resolver.html#managing-value-resolvers.

**Comment by Xesau** [comment]
@MatTheCat That is true, but that is not my point. The documentation (https://symfony.com/doc/current/controller.html, as well as the page you link) teaches us that we can obtain the Request object (through the request value resolver) by specifying just a Request argument in the controller function. In my opinion, the fact that resolution is attempted by the Doctrine entity resolver before the Request is unexpected behavior. One should not be required to specify what is assumed to be the default. The current configuration is therefore not sensible and should be changed. Or the documentation should explicity state that you should never rely on the type annotation alone and always specify which resolver should be used.

**Comment by MatTheCat** [comment]
I was not making a point, just giving you a short-term solution.

`EntityValueResolver`’s priority is so high because it must be greater than `RequestAttributeValueResolver`’s, but this already triggered some other changes (like #48032).

I cannot think of use-cases where it would be problematic to increase the `RequestValueResolver`’s priority; will you open a PR?

**Comment by smnandre** [comment]
There was one, not sure if that is still the case: https://github.com/doctrine/DoctrineBundle/pull/1554#discussion_r949478887

**Comment by MatTheCat** [comment]
@smnandre the comment you linked mentions the `RequestAttributeValueResolver`, not `RequestValueResolver`.

It actually explains why “`EntityValueResolver`’s priority […] must be greater than `RequestAttributeValueResolver`’s” :grin:

**Comment by smnandre** [comment]
#SimonPleaseTalkLessAndSleepMoreS07E07 .... 😶‍🌫️

**Comment by nicolas-grekas** [comment]
> 40–50ms

That's a lot and that's surprising. Normally this should be fast because this should all be just about loading pregenerated code.
Can you please try to benchmark what takes so long? Maybe you're missing some metadata cache or something else?

**Comment by stof** [comment]
@nicolas-grekas metadata are cached for entities. However, the cache is not marked as authoritative so that any cache miss is treated as having no metadata without calling the driver again.
When trying to detect whether the argument is an entity, doing such check on the `Request` or `Session` classes trigger such cache miss.

However, this value looks like a lot to me (unless you consider that it includes the whole instantiation of the ORM object graph, but this would be a valid assumption only if *nothing* else in the request handling would use the ORM, so that this instantiation would not be moved later in the request processing otherwise)
